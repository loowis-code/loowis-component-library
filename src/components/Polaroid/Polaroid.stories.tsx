import type { Meta, StoryObj } from "@storybook/react";
import Polaroid from "./Polaroid";

const meta: Meta<typeof Polaroid> = {
  title: "Components/Polaroid",
  component: Polaroid,
};

export default meta;

type Story = StoryObj<typeof Polaroid>;

export const SquareWithCaption: Story = {
    args: {
        filename: '/picture.jpg',
        caption: 'Lewis Inches',
    }
};

export const SquareWithCaptionAndLinks: Story = {
    args: {
        filename: '/picture.jpg',
        caption: 'Lewis Inches',
        links: [
            { href: '/link1', text: 'Link 1' },
            { href: '/link2', text: 'Link 2' },
            { href: '/link3', text: 'Link 3' },
        ],
    }
};

export const SquareWithoutCaption: Story = {
    args: {
        filename: '/picture.jpg',
    }
};

export const SixteenByNineWithCaption: Story = {
    args: {
        filename: '/landscape.webp',
        caption: 'Lewis Inches',
    }
};

export const SixteenByNineWithoutCaption: Story = {
    args: {
        filename: '/landscape.webp',
    }
};

export const NineBySixteenWithCaption: Story = {
    args: {
        filename: '/portrait.webp',
        caption: 'Lewis Inches',
    }
};

export const NineBySixteenWithoutCaption: Story = {
    args: {
        filename: '/portrait.webp',
    }
};