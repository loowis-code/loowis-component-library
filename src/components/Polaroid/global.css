@font-face {
    font-family: josefin-regular;
    src: url("../../../shared/josefin/JosefinSans-Regular.ttf");
}

.polaroidFrame {
    background-color: #fff;
    padding: 2rem 1rem 2.5rem;
    width: 20rem;
    height: auto;
    text-align: left;
    border: 0.25rem solid #000;
    box-shadow: 0.25rem 0.25rem #000;
    padding-bottom: 5rem;
    position: relative;
    justify-content: center;
    align-items: center;
    text-align: center;
}

.polaroidImage {
    width: auto;
    max-width: 100%;
    height: auto;
}

.polaroidLinks {
    display: flex;
}

.polaroidName {
    background-color: #ffdb58;
    border: #e6c44c 0.25rem solid;
    box-shadow: #e6c44c 0.25rem 0.25rem;
    width: fit-content;
    margin-top: 0.5rem;
    padding: 0.5rem;
    font-family: josefin-regular;
    position: absolute;
    bottom: 5%;
    left: 50%;
    transform: translateX(-50%);
}