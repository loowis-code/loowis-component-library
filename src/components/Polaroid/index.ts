import Polaroid from './Polaroid';

export default Polaroid;