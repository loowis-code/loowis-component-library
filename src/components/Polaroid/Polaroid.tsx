import './global.css';
import Image from 'next/image';
import Button from '../Button';

type PolaroidProps = {
    filename: string;
    alt?: string;
    caption?: string;
    links?: { href: string, text: string }[];
};

export default function Polaroid(props: PolaroidProps) {
    return (
        <figure className='polaroidFrame' data-testid='polaroid'>
                <Image src={props.filename} alt={props.alt} className='polaroidImage' height={300} width={300}/>
                {props.links && 
                    <div className='polaroidLinks'>
                        {props.links.map((link, index) => (
                            <Button key={index} buttonLink={link.href} buttonText={link.text}/>
                        ))}
                    </div>
                }
                {props.caption && <figcaption className='polaroidName'>{props.caption}</figcaption>}


        </figure>
    );
}