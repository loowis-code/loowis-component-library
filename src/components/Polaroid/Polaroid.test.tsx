import Polaroid from './Polaroid';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

describe('Header', () => {
    it('renders correctly', () => {
        const tree = render(<Polaroid filename='/picture.jpg' caption='Test Caption' links={[{href:'test-link', text:'Test Link'}]}/>);
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly without links', () => {
        const tree = render(<Polaroid filename='/picture.jpg' caption='Test Caption'/>);
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly without caption and with links', () => {
        const tree = render(<Polaroid filename='/picture.jpg' links={[{href:'test-link', text:'Test Link'}]}/>);
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly without caption and links', () => {
        const tree = render(<Polaroid filename='/picture.jpg'/>);
        expect(tree).toMatchSnapshot();
    });
});