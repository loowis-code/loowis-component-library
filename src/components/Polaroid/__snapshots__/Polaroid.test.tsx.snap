// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <figure
        class="polaroidFrame"
        data-testid="polaroid"
      >
        <img
          class="polaroidImage"
          data-nimg="1"
          decoding="async"
          height="300"
          loading="lazy"
          src="/_next/image?url=%2Fpicture.jpg&w=640&q=75"
          srcset="/_next/image?url=%2Fpicture.jpg&w=384&q=75 1x, /_next/image?url=%2Fpicture.jpg&w=640&q=75 2x"
          style="color: transparent;"
          width="300"
        />
        <div
          class="polaroidLinks"
        >
          <a
            class="buttonLink"
            data-testid="button"
            href="test-link"
          >
            Test Link
          </a>
        </div>
        <figcaption
          class="polaroidName"
        >
          Test Caption
        </figcaption>
      </figure>
    </div>
  </body>,
  "container": <div>
    <figure
      class="polaroidFrame"
      data-testid="polaroid"
    >
      <img
        class="polaroidImage"
        data-nimg="1"
        decoding="async"
        height="300"
        loading="lazy"
        src="/_next/image?url=%2Fpicture.jpg&w=640&q=75"
        srcset="/_next/image?url=%2Fpicture.jpg&w=384&q=75 1x, /_next/image?url=%2Fpicture.jpg&w=640&q=75 2x"
        style="color: transparent;"
        width="300"
      />
      <div
        class="polaroidLinks"
      >
        <a
          class="buttonLink"
          data-testid="button"
          href="test-link"
        >
          Test Link
        </a>
      </div>
      <figcaption
        class="polaroidName"
      >
        Test Caption
      </figcaption>
    </figure>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Header renders correctly without caption and links 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <figure
        class="polaroidFrame"
        data-testid="polaroid"
      >
        <img
          class="polaroidImage"
          data-nimg="1"
          decoding="async"
          height="300"
          loading="lazy"
          src="/_next/image?url=%2Fpicture.jpg&w=640&q=75"
          srcset="/_next/image?url=%2Fpicture.jpg&w=384&q=75 1x, /_next/image?url=%2Fpicture.jpg&w=640&q=75 2x"
          style="color: transparent;"
          width="300"
        />
      </figure>
    </div>
  </body>,
  "container": <div>
    <figure
      class="polaroidFrame"
      data-testid="polaroid"
    >
      <img
        class="polaroidImage"
        data-nimg="1"
        decoding="async"
        height="300"
        loading="lazy"
        src="/_next/image?url=%2Fpicture.jpg&w=640&q=75"
        srcset="/_next/image?url=%2Fpicture.jpg&w=384&q=75 1x, /_next/image?url=%2Fpicture.jpg&w=640&q=75 2x"
        style="color: transparent;"
        width="300"
      />
    </figure>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Header renders correctly without caption and with links 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <figure
        class="polaroidFrame"
        data-testid="polaroid"
      >
        <img
          class="polaroidImage"
          data-nimg="1"
          decoding="async"
          height="300"
          loading="lazy"
          src="/_next/image?url=%2Fpicture.jpg&w=640&q=75"
          srcset="/_next/image?url=%2Fpicture.jpg&w=384&q=75 1x, /_next/image?url=%2Fpicture.jpg&w=640&q=75 2x"
          style="color: transparent;"
          width="300"
        />
        <div
          class="polaroidLinks"
        >
          <a
            class="buttonLink"
            data-testid="button"
            href="test-link"
          >
            Test Link
          </a>
        </div>
      </figure>
    </div>
  </body>,
  "container": <div>
    <figure
      class="polaroidFrame"
      data-testid="polaroid"
    >
      <img
        class="polaroidImage"
        data-nimg="1"
        decoding="async"
        height="300"
        loading="lazy"
        src="/_next/image?url=%2Fpicture.jpg&w=640&q=75"
        srcset="/_next/image?url=%2Fpicture.jpg&w=384&q=75 1x, /_next/image?url=%2Fpicture.jpg&w=640&q=75 2x"
        style="color: transparent;"
        width="300"
      />
      <div
        class="polaroidLinks"
      >
        <a
          class="buttonLink"
          data-testid="button"
          href="test-link"
        >
          Test Link
        </a>
      </div>
    </figure>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Header renders correctly without links 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <figure
        class="polaroidFrame"
        data-testid="polaroid"
      >
        <img
          class="polaroidImage"
          data-nimg="1"
          decoding="async"
          height="300"
          loading="lazy"
          src="/_next/image?url=%2Fpicture.jpg&w=640&q=75"
          srcset="/_next/image?url=%2Fpicture.jpg&w=384&q=75 1x, /_next/image?url=%2Fpicture.jpg&w=640&q=75 2x"
          style="color: transparent;"
          width="300"
        />
        <figcaption
          class="polaroidName"
        >
          Test Caption
        </figcaption>
      </figure>
    </div>
  </body>,
  "container": <div>
    <figure
      class="polaroidFrame"
      data-testid="polaroid"
    >
      <img
        class="polaroidImage"
        data-nimg="1"
        decoding="async"
        height="300"
        loading="lazy"
        src="/_next/image?url=%2Fpicture.jpg&w=640&q=75"
        srcset="/_next/image?url=%2Fpicture.jpg&w=384&q=75 1x, /_next/image?url=%2Fpicture.jpg&w=640&q=75 2x"
        style="color: transparent;"
        width="300"
      />
      <figcaption
        class="polaroidName"
      >
        Test Caption
      </figcaption>
    </figure>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
